PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ["auth_totp.device"]
-    _inherit = ['res.config.settings']
-    _inherit = ['res.users']

Total Changes: 7

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/auth_totp_mail/models/auth_totp_device.py
+++ b/addons/auth_totp_mail/models/auth_totp_device.py
@@ -5,7 +5,7 @@ from collections import defaultdict
 
 
 class Auth_TotpDevice(models.Model):
-    _inherit = ["auth_totp.device"]
+    _inherit = "auth_totp.device"
 
     def unlink(self):
         """ Notify users when trusted devices are removed from their account. """

--- a/addons/auth_totp_mail/models/auth_totp_rate_limit_log.py
+++ b/addons/auth_totp_mail/models/auth_totp_rate_limit_log.py
@@ -2,6 +2,7 @@ from odoo import fields, models
 
 
 class AuthTotpRateLimitLog(models.TransientModel):
+    _name = 'auth.totp.rate.limit.log'
     _description = 'TOTP rate limit logs'
 
     def init(self):

--- a/addons/auth_totp_mail/models/res_config_settings.py
+++ b/addons/auth_totp_mail/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     auth_totp_enforce = fields.Boolean(
         string="Enforce two-factor authentication",

--- a/addons/auth_totp_mail/models/res_users.py
+++ b/addons/auth_totp_mail/models/res_users.py
@@ -21,7 +21,7 @@ TOTP_RATE_LIMITS = {
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     def write(self, vals):
         res = super().write(vals)
